controller/user/manager.py

```diff
@@ -101,3 +101,49 @@ def get_active_users(minutes: str, order_by_interaction: bool) -> User:
 @param_throttle(seconds=10)
 def update_last_interaction(user_id: str) -> None:
     user_activity.update_last_interaction(user_id)
+
+
+def get_mapped_sorted_paginated_users(
+    active_users: Dict[str, Any],
+    sort_key: str,
+    sort_direction: int,
+    offset: int,
+    limit: int,
+) -> List[Dict[str, str]]:
+
+    final_users = []
+
+    # mapping users with the users in kratos
+    active_users_ids = list(active_users.keys())
+
+    for user_id in active_users_ids:
+        get_user = kratos.__get_identity(user_id, False)["identity"]
+        if get_user:
+            get_user["email"] = get_user["traits"]["email"]
+            get_user["verified"] = get_user["verifiable_addresses"][0]["verified"]
+            active_user_by_id = active_users[user_id]
+            get_user["last_interaction"] = active_user_by_id["lastInteraction"]
+            get_user["role"] = active_user_by_id["role"]
+            get_user["organization"] = active_user_by_id["organizationName"]
+
+            final_users.append(get_user)
+
+    final_users = sorted(
+        final_users,
+        key=lambda x: x[sort_key] if sort_key in x else None,
+        reverse=(
+            sort_direction == -1
```

Comment:
could be the error mentioned in the comments: switch 1 and -1 around here

Reply:
no, this is not the reason, this would revert the whole array, the only problem is sorting with numbers

Reply:
changed it in the end :)